### PR TITLE
Feature/antiraid setup cmd

### PR DESCRIPTION
**Pull request by hburn7**

### CODE REVIEW

File: Kaguya/Kaguya/Discord/Commands/Configuration/AntiraidSetup.cs [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+            _kaguyaServerRepository = kaguyaServerRepository;
+            _antiraidConfigRepository = antiraidConfigRepository;
+            _commonEmotes = commonEmotes;
+        }
+
+        [Command(RunMode = RunMode.Async)]
+        [Summary("Displays an interactive setup utility for configuring the Kaguya Antiraid service.")]
+        public async Task AntiraidSetupCommand()
+        {
+            /* todo: Ensure more than 1 setup cannot run in a given server at a time. */
```
Comment (hburn7) [inline Kaguya/Kaguya/Discord/Commands/Configuration/AntiraidSetup.cs]: @BitMasher We need to somehow check to make sure the same server hasn't executed this command more than once while a current setup is running (i.e. before the "stages" below are finished). We would somehow cache the server id into a thread-safe collection somewhere in the program, then remove the id once the "stages" are complete. If the server id is in the collection, we send an error.

What is the best way to complete this?

Comment (BitMasher) [reply Kaguya/Kaguya/Discord/Commands/Configuration/AntiraidSetup.cs]: a concurrentdictionary to track setup state could work

File: Kaguya/Kaguya/Discord/CommonEmotes.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                        _logger.LogWarning("Emote server could not be found. Some emotes will not be populated.");
+                    }
+
+                    _redCrossEmote = emoteGuild?.Emotes.FirstOrDefault(x => x.Name.ToLower() == "redcross");
+
+                    if (_redCrossEmote == null)
+                    {
+                        _logger.LogWarning("Redcross emote not populated. Using default ❌ emote.");
+                        _redCrossEmote = new Emoji("❌");
+                    }
```
Comment (BitMasher) [inline Kaguya/Kaguya/Discord/CommonEmotes.cs]: refactor this into a private method, i suspect you'll be doing this a lot more than once to get all the other emojis

**Review by BitMasher — CHANGES_REQUESTED** [review]

**Comment by hburn7** [comment]
Made all changes requested by BitMasher. Merging.

**Pull request merged**